packages/application-generic/src/services/cron/cron.service.ts

```diff
@@ -0,0 +1,283 @@
+      Logger.verbose(
+        `The '${this.cronServiceName}' CRON service has no active jobs and will not start up`,
+        LOG_CONTEXT
+      );
+
+      return;
+    } else {
+      Logger.log(
+        `Active CRON jobs found for ${
+          this.cronServiceName
+        }: [${this.activeJobs.join(', ')}]`,
+        LOG_CONTEXT
+      );
+    }
+
+    try {
+      Logger.log(
+        `Starting the '${this.cronServiceName}' CRON service up`,
+        LOG_CONTEXT
+      );
+      await Promise.race([
+        this.initialize(),
+        new Promise((resolve, reject) =>
+          setTimeout(
+            () =>
+              reject(
+                `Timed out while starting the ${this.cronServiceName} CRON service`
+              ),
+            2000
+          )
+        ),
+      ]);
+      await this.createSendCronMetricsJob();
+      Logger.log(
+        `Starting up the '${this.cronServiceName}' CRON service has finished`,
+        LOG_CONTEXT
+      );
+    } catch (error) {
+      Logger.error(
+        `Failed to start the '${this.cronServiceName}' CRON service`,
+        error,
+        LOG_CONTEXT
+      );
+    }
+  }
+
+  /**
+   * This method is called when the application is shutting down.
+   * It ensures that all relevant jobs are removed from the CRON service before shutting down.
+   */
+  async onApplicationShutdown() {
+    if (!this.activeJobs || this.activeJobs.length === 0) {
+      Logger.verbose(
+        `The '${this.cronServiceName}' CRON service has no active jobs and will not shut down`,
+        LOG_CONTEXT
+      );
+
+      return;
+    }
+
+    try {
+      Logger.log(
+        `Shutting the '${this.cronServiceName}' CRON service down`,
+        LOG_CONTEXT
+      );
+      await this.shutdown();
+      Logger.log(
+        `Shutting down the '${this.cronServiceName}' CRON service has finished`,
+        LOG_CONTEXT
+      );
+    } catch (error) {
+      Logger.error(
+        `Failed to shut down the '${this.cronServiceName}' CRON service`,
+        error,
+        LOG_CONTEXT
+      );
+    }
+  }
+
+  private isActiveJob(jobName: JobCronNameEnum): boolean {
+    return this.activeJobs.includes(jobName);
+  }
+
+  public async add<TData = unknown>(
+    jobName: JobCronNameEnum,
+    processor: (job: CronJobData<TData>) => Promise<void>,
+    interval: string,
+    options: CronOptions
+  ): Promise<void> {
+    if (!this.isActiveJob(jobName)) {
+      Logger.verbose(
+        `The '${jobName}' job is not active and will not be added to the CRON service`,
+        LOG_CONTEXT
+      );
+
+      return;
+    }
+
+    const combinedOptions = {
+      ...DEFAULT_CRON_OPTIONS,
+      ...options,
+    };
+    this.handleJobOutcome(jobName, 'create-started');
+    try {
+      // eslint-disable-next-line @typescript-eslint/no-this-alias
+      const _this = this;
+      this.addJob(
+        jobName,
+        async function runCronJob(job) {
+          nr.startBackgroundTransaction(
```

Comment:
Strong transaction monitoring is in place for all running cron jobs, following the queues transaction monitoring pattern.